core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -266,10 +266,12 @@ private[spark] class ExecutorAllocationManager(
       // executors and inform the cluster manager to cancel the extra pending requests
       val oldNumExecutorsTarget = numExecutorsTarget
       numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
+      if (numExecutorsTarget < oldNumExecutorsTarget) {
```

Comment:
@ryan-williams that may be so, but it's much simpler to reason about if we just set `numExecutorsToAdd = 1` every time our target exceeds the number needed. I would just do it outside for that reason if it doesn't change any functionality.